userguide/storage.rst

```diff
@@ -713,53 +713,70 @@ existing datasets to close them--the remaining
+of any data that is repeated in different blocks or files, rather than storing 
+identical copies each time the data occurs. Deduplication 
+potentially allows repeated and identical data to be stored very 
+efficiently in a smaller space, but also places a 
+very heavy burden on RAM and on the CPU. In particular, dedup 
+significantly reduces write performance and usually requires large (and at 
+times huge) amounts of RAM.  See :ref:`Deduplication RAM requirements` for details.
+Because of the demands that dedup places on a system and the impact on performance, 
+enabling dedup is **usually not recommended** and the setting should be left in
+the default setting ("off") in most situations.
+
+.. warning::
+
+   The more data written to a deduplicated dataset, the more RAM it 
+   requires. The DDTs (dedup tables) also compete with other data for RAM,
+   and can be pushed out of RAM in some cases. When the system starts 
+   storing DDTs on disk because they no longer fit into RAM or have been
+   pushed out of RAM, performance craters.
+   
+   Importing an unclean deduplicated pool can also require large amounts of RAM, 
+   so that the deduplicated data in the pool can be checked during import. 
+   If the system does not have the needed RAM, it will panic. The only 
+   solution then is to add more RAM or recreate the pool.
+
+   **There is also no quick way to undedup data within a dataset once deduplication 
+   is enabled, since disabling deduplication DOES NOT CHANGE existing data.** The 
+   simplest way to undedup existing data is to copy the deduped data to a new dataset 
+   or location that does not have dedup set. The data will be undeduped when written. 
+   The simplest way to undedup data and keep it in the same pool, zvol or dataset is 
+   as follows: disable dedup on the pool, zvol or dataset, then copy the data within 
+   the same pool, zvol or dataset. Once the copy is confirmed to be correct, the 
+   original data can optionally be deleted. If the pool does not have enough free space
+   for this procedure, then disable dedup, copy the data to another system or storage
+   device, check the copies are correct and safe, then delete the original data, and 
+   transfer back the copied data to the pool.
+
+   For these reasons, **think carefully before enabling dedup!** 
+   Deduplication is **usually not recommended unless the data contains a 
+   very substantial level of duplication, and the system (especially 
+   RAM) can cope acceptably with the additional burden**. In most cases, 
+   compression provides storage gains comparable to deduplication with 
+   less impact on performance.  This `article 
+   <http://constantin.glez.de/blog/2011/07/zfs-dedupe-or-not-dedupe>`_ 
+   provides a good description of the value versus cost considerations 
+   for deduplication. Unless a lot of RAM and a lot of highly duplicate 
+   data is available, **do not change the default deduplication setting 
+   of "Off"!**  
+
```

Comment:
Get rid of "For these reasons" Rewrite to get rid of the aside. 
Rewrite to use directions/statements of fact.

Reply:
Tried rewording it in line with comments. Please review.

Reply:
Tried rewording it in line with comments. Please review.